include/mbedtls/config.h

```diff
@@ -29,9 +29,20 @@
 #ifndef MBEDTLS_CONFIG_H
 #define MBEDTLS_CONFIG_H
 
-#if defined(_MSC_VER) && !defined(_CRT_SECURE_NO_DEPRECATE)
+#ifdef _MSC_VER
+#ifndef _CRT_SECURE_NO_DEPRECATE
 #define _CRT_SECURE_NO_DEPRECATE 1
 #endif
+#include <winsock2.h>
+#include <tchar.h>
+#define MBEDTLS__T _T
+#define MBEDTLS_TCHAR TCHAR
+#define MBEDTLS_FOPEN _tfopen
+#else
+#define MBEDTLS__T(x) x
+#define MBEDTLS_TCHAR char
+#define MBEDTLS_FOPEN fopen
```

Comment:
Note that #1029 also defines an abstraction for `fopen`. This and #1029 will need to be reconciled.

Reply:
Yes, but code in this PR is already working, and #1029 is still in development, and Windows-specific part of it probably was never tested.